### PR TITLE
special render packages should render with materials colors only

### PR DESCRIPTION
**Pull request by mjkkirschner**
Screen Shot 2020-03-19 at 3 16 47 PM
add test
update shader

### Purpose

enable the `SpecialRenderPackage` flag and use it in the shader.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Watch3D/shaderSource/psDynamoMesh.hlsl [hunk 1]
```diff
@@ -69,6 +69,11 @@ float4 main(PSInputCustom input, bool isFrontFacing : SV_IsFrontFace) : SV_Targe
     //https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-semantics
 	 input.n = isFrontFacing ? input.n : -input.n;
 
+	 // if this is a special render package - it should render with the material colors, ambient light
+	 // and not be directionally lit.
+	 if (isSpecialRenderPackage) {
```
Comment (mmisol) [inline src/DynamoCoreWpf/ViewModels/Watch3D/shaderSource/psDynamoMesh.hlsl]: In line 45 it says
```
IsSpecialRenderPackage = 8, //not used
```
Should we remove the `//not used` part now?

Comment (mjkkirschner) [reply src/DynamoCoreWpf/ViewModels/Watch3D/shaderSource/psDynamoMesh.hlsl]: thanks @mmisol will do.

**Comment by mjkkirschner** [comment]
Believe all comments regarding this flag are now up to date.

**Comment by aparajit-pratap** [comment]
Looks prettier :)

**Comment by aparajit-pratap** [comment]
Was there any visual change apart from the gizmo since the directional light, grid and grid axes are also special render packages?

**Comment by mjkkirschner** [comment]
@aparajit-pratap there should not be, as 
1. the light is not a render package AFAIK - it has no geometry
2. grid and axes are lines which don't use this shader.

**Comment by aparajit-pratap** [comment]
In that case, I wonder, why we are doing this: https://github.com/DynamoDS/Dynamo/blob/1464172dd3a1528deca7746a226cd5fabf0dbe13/src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs#L1226

**Comment by mjkkirschner** [comment]
@aparajit-pratap yep, not sure, it doesn't seem to make sense, it's not geometry and it's not a DynamoGeometryModel3d - so this likely doesn't have much of an effect anywhere...

**Comment by mjkkirschner** [comment]
@aparajit-pratap it's possible it might make a difference in the sorting algorithm - https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs#L2240

but even there, you wouldn't need to sort a light... in any case, this should only effect DynamoModelGeometry3d objects of which none of these (light,grid,axes) are AFIK.

**Comment by aparajit-pratap** [comment]
LGTM

**Pull request merged**